qbit-http-storages/src/commonMain/kotlin/qbit.storage/YandexDiskStorage.kt

```diff
@@ -0,0 +1,250 @@
+            println(e)
+        }
+        val json = Json.indented.parseJson(getUrlToUploadFile(client, paths[paths.size - 1], key.name).readText())
+        var urlToUploadFile = (json as JsonObject).getPrimitive("href").content
+        var filePutResponse = uploadFile(client, urlToUploadFile, value)
+        client.close()
+    }
+
+    @InternalCoroutinesApi
+    override fun overwrite(key: Key, value: ByteArray) = runBlocking {
+        val client = HttpClient()
+        val paths = generatePutResourcePaths(key.ns)
+        val json = Json.indented.parseJson(getUrlToUploadFile(client, paths[paths.size - 1], key.name).readText())
+        var urlToUploadFile = (json as JsonObject).getPrimitive("href").content
+        var filePutResponse = uploadFile(client, urlToUploadFile, value)
+        client.close()
+    }
+
+    @InternalCoroutinesApi
+    override fun load(key: Key): ByteArray? = runBlocking {
+        val filePath = getFullPath(key)
+
+        val client = HttpClient()
+        val response = getUrlToDownload(client, filePath)
+        val json = Json.indented.parseJson(response.readText())
+        val downloadUrl = (json as JsonObject).getPrimitive("href").content
+
+        val byteArray = getFile(client, downloadUrl)
+        client.close()
+        byteArray
+    }
+
+    @InternalCoroutinesApi
+    override fun keys(namespace: Namespace): Collection<Key> = runBlocking {
+        var fullPath = getFullPath(namespace)
+
+        val client = HttpClient()
+        val response = client.get<HttpResponse>(YANDEX_DISK_API_RESOURCES) {
+            parameter("path", fullPath)
+            header("Authorization", "OAuth $accessToken")
+        }
+        val json = Json(JsonConfiguration.Stable)
+        val resource = json.parse(Resource.serializer(), response.readText())
+        val fileNames = getFileNamesInResourceByType(resource, "file")
+        val keys = wrapFileNamesToKeys(fileNames, namespace)
+        client.close()
+        keys
+    }
+
+    @InternalCoroutinesApi
+    override fun subNamespaces(namespace: Namespace): Collection<Namespace> = runBlocking {
+        var fullPath = getFullPath(namespace)
+
+        val client = HttpClient()
+        val response = client.get<HttpResponse>(YANDEX_DISK_API_RESOURCES) {
+            parameter("path", fullPath)
+            header("Authorization", "OAuth $accessToken")
+        }
+        val json = Json(JsonConfiguration.Stable)
+        val resource = json.parse(Resource.serializer(), response.readText())
+        val dirNames = getFileNamesInResourceByType(resource, "dir")
+        val namespaces = wrapDirNamesToNamespaces(dirNames, namespace)
+        client.close()
+        namespaces
+    }
+
+    @InternalCoroutinesApi
+    override fun hasKey(key: Key): Boolean = runBlocking {
+        var fullPath = getFullPath(key)
+
+        val client = HttpClient()
+        try {
+            val response = client.get<String>(YANDEX_DISK_API_RESOURCES) {
```

Comment:
You are anyway do not use reponse as string, can you get response as HttpResponse and check it response code instead of try-catch?